functions/api/helpers.js

```diff
@@ -20,4 +20,68 @@ async function getUserDisplayName(userRef) {
   return userRef.split('users/')[1];
 }
 
-export default getUserDisplayName;
+/** Update the KPI progression collection with at most one value each
+ * day. Check for any existing values on specified `date` and either
+ * add new or update existing measurement value.
+ *
+ * `kpiRef` is the reference to the parent KPI to update.
+ * `date` specifies which date the measurement is made.
+ * `value` is the progression value to set.
+ */
+export async function updateKPIProgressionValue(kpiRef, date, value) {
+  date = startOfDay(date);
+
+  const progressCollectionRef = kpiRef.collection('progress');
+
+  const existingValueRef = await progressCollectionRef
+    .orderBy('timestamp', 'desc')
+    .where('timestamp', '>=', date)
+    .where('timestamp', '<=', endOfDay(date))
+    .limit(1)
+    .get()
+    .then((snapshot) => (!snapshot.empty ? snapshot.docs[0].ref : null));
```

Comment:
Parentesene i returuttrykket har vel ingen funksjon?

Reply:
Er det ikke litt pent da? 
<img width="712" alt="image" src="https://user-images.githubusercontent.com/2866225/195870242-c98bb1d5-f6cd-46da-8660-83dbcf085154.png">

Reply:
Hjernen min begynner da å lete etter hvilken presedensregel de er der for å overstyre. :stuck_out_tongue: Men det er ikke så farlig...